crates/miden-tx/src/host/mod.rs

```diff
@@ -519,27 +521,102 @@ where
         Ok(())
     }
 
-    /// Aborts the transaction by extracting the
-    /// [`TransactionSummary`](miden_objects::transaction::TransactionSummary) from the stack and
-    /// returns it in an error.
+    // TRANSACTION SUMMARY HELPER
+    // --------------------------------------------------------------------------------------------
+
+    /// Builds a [TransactionSummary] by extracting data from the advice provider and validating
+    /// commitments against the host's state.
+    pub(crate) fn build_tx_summary(
```

Comment:
nit: I would move this down to the `HELPER FUNCTIONS` section (i.e., after line 623).